packages/core/src/storage/StorageUtility.ts

```diff
@@ -175,13 +189,23 @@ export default class StorageUtility implements IStorageUtility {
         }
         return parsedObject;
       } catch (err) {
+        let invalidObject;
         if (options.userId) {
-          await this.deleteForUser(options.userId, key, {
+          invalidObject = await this.getForUser(options.userId, key, {
             secure: options.secure,
           });
         } else {
-          await this.delete(key, { secure: options.secure });
+          invalidObject = await this.get(key, {
+            secure: options.secure,
+          });
         }
+        throw new Error(
+          `Object ${JSON.stringify(
+            invalidObject
+          )} does not match expected schema ${JSON.stringify(
+            options.schema
+          )}: ${err.toString()}. Please clear your local storage.`
```

Comment:
I'm not sure how this message would actually appear in a console or log file, but I'm thinking we'd need a newline before the `Please clear your local storage`...?